llvm-build/llvm.cmake

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+cmake_minimum_required(VERSION 3.4 FATAL_ERROR)
+
+message(STATUS CMAKE_HOST_SYSTEM_NAME=${CMAKE_HOST_SYSTEM_NAME})
+
+if (${CMAKE_HOST_SYSTEM_NAME} MATCHES "Windows")
+  find_program(SCCACHE sccache)
+  if(SCCACHE)
+      set(LLVM_CCACHE_BUILD OFF CACHE BOOL "")
+      set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${SCCACHE}")
+      message(STATUS RULE_LAUNCH_COMPILE=${RULE_LAUNCH_COMPILE})
+      set(CMAKE_C_COMPILER_LAUNCHER "sccache" CACHE STRING "")
+      set(CMAKE_CXX_COMPILER_LAUNCHER "sccache" CACHE STRING "")
+  else()
+    message(STATUS "Not using sccache")
+  endif()
+else()
+  # Prepare cache if we find it
+  find_program(CCACHE ccache)
+  if(CCACHE)
+    message(STATUS "Using CCache for linux/darwin")
+    set(LLVM_CCACHE_BUILD ON CACHE BOOL "")
+    set(LLVM_CCACHE_DIR $ENV{CCACHE_DIR} CACHE STRING "")
+    set(LLVM_CCACHE_MAXSIZE "2G" CACHE STRING "")
+    message(STATUS CCACHE_DIR=$ENV{CCACHE_DIR})
+    message(STATUS LLVM_CCACHE_BUILD=${LLVM_CCACHE_BUILD})
+    message(STATUS LLVM_CCACHE_DIR=${LLVM_CCACHE_DIR})
+    message(STATUS LLVM_CCACHE_MAXSIZE=${LLVM_CCACHE_MAXSIZE})
+  else()
+    message(STATUS "Not using CCache")
+  endif()
+endif()
+
+
+set(CPACK_PACKAGE_FILE_NAME $ENV{PKG_NAME} CACHE STRING "")
+message(STATUS CPACK_PACKAGE_FILE_NAME=${CPACK_PACKAGE_FILE_NAME})
+
+# Set up main build props
+
+set(CMAKE_BUILD_TYPE MinSizeRel CACHE STRING "")
+
+set(LLVM_TARGETS_TO_BUILD "X86" CACHE STRING "")
+
+set(PACKAGE_VENDOR LLVM.org CACHE STRING "")
+
+# Turn off
+set(LLVM_ENABLE_ASSERTIONS OFF CACHE BOOL "")
+set(LLVM_BUILD_EXAMPLES OFF CACHE BOOL "")
+set(LLVM_ENABLE_RTTI OFF CACHE BOOL "")
+
+# Packing
+set(CPACK_BINARY_DEB OFF CACHE BOOL "")
+set(CPACK_BINARY_FREEBSD OFF CACHE BOOL "")
+set(CPACK_BINARY_IFW OFF CACHE BOOL "")
+set(CPACK_BINARY_NSIS OFF CACHE BOOL "")
+set(CPACK_BINARY_RPM OFF CACHE BOOL "")
+set(CPACK_BINARY_STGZ OFF CACHE BOOL "")
+set(CPACK_BINARY_TBZ2 OFF CACHE BOOL "")
+set(CPACK_BINARY_TXZ OFF CACHE BOOL "")
+set(CPACK_BINARY_TZ OFF CACHE BOOL "")
+
+if (${CMAKE_HOST_SYSTEM_NAME} MATCHES "Windows")
+  message(STATUS "Configuring for Windows")
+  set(LLVM_USE_CRT_RELEASE "MT" CACHE STRING "")
+  set(LLVM_USE_CRT_MINSIZEREL "MT" CACHE STRING "")
+  set(LLVM_BUILD_LLVM_C_DYLIB ON CACHE BOOL "")
+  set(CMAKE_INSTALL_UCRT_LIBRARIES ON CACHE BOOL "")
+  set(CPACK_BINARY_ZIP ON CACHE BOOL "")
+  set(LLVM_DYLIB_COMPONENTS "core;debuginfodwarf;linker;support;target;bitwriter" CACHE STRING "")
+else()
+  set(LLVM_BUILD_LLVM_DYLIB ON CACHE BOOL "")
+  set(CPACK_BINARY_TGZ ON CACHE BOOL "")
+  set(LLVM_DYLIB_COMPONENTS "core;debuginfodwarf;linker;support;target;bitwriter" CACHE STRING "")
+endif()
```

Comment:
Both `if` and `else` parts end with command `set(LLVM_DYLIB_COMPONENTS "core;debuginfodwarf;linker;support;target;bitwriter" CACHE STRING "")`. Looks like the command can be extracted and placed after the line 76.

It is suspicious that the Win-specific part (`if (${CMAKE_HOST_SYSTEM_NAME} MATCHES "Windows")`) contains the fragments `DYLIB` (in LLVM_BUILD_LLVM_C_**DYLIB**, LLVM_**DYLIB**_COMPONENTS ).

Reply:
In this case, `DYLIB` refers to dynamic library as a general term, and it is an option exposed by the CMakeLists for LLVM. See https://github.com/llvm/llvm-project/blob/ebd8eee62a431a6744c3f187fcda58e5dea08499/llvm/CMakeLists.txt#L641.